File: 1496-lucky-numbers-in-a-matrix/lucky-numbers-in-a-matrix.py
class Solution:
    def luckyNumbers (self, matrix: List[List[int]]) -> List[int]:
        row = [float("inf")]*len(matrix)
        col = [float("-inf")]*len(matrix[0])
        for i in range(len(matrix)):
            for y in range(len(matrix[0])):
                col[y] = max(col[y],matrix[i][y])
                row[i] = min(row[i],matrix[i][y])
       
        dic = Counter(col)
        ans = []
        for x in row:
            if x in dic and dic[x] > 0:
                ans.append(x)
                dic[x] -= 1
        
        return ans

File: 1496-lucky-numbers-in-a-matrix/lucky-numbers-in-a-matrix.go
import "fmt"
func luckyNumbers (matrix [][]int) []int {
    row := make([]int,len(matrix[0]))
    col := make([]int, len(matrix))
    for i,_ := range matrix{
        for j,val := range matrix[i]{
            if row[j] < val || row[j] == 0{
                row[j] = val
            }
            if col[i] > val || col[i] == 0{
                col[i] = val
            }
        }
    }
    ans := []int{}
 
    for _,val := range row{
        for _,vall := range col{
            if val == vall{
                ans = append(ans,val)
            }
        }
    }
    return ans
}